src/renderer/AutoEQ.tsx

```diff
@@ -1,75 +1,117 @@
+const AutoEQ = () => {
+  const NO_DEVICE_SELECTION = 'Pick a device first! 🎧';
+  const NO_RESPONSES = 'No supported responses 😞';
+  const NO_RESPONSE_SELECTION = 'Pick a response! 🔊';
 
-const TARGET_FREQUENCY_RESPONSE = [
-  'IEF Neutral',
-  'Diffuse Field',
-  'Free Field',
-  'Harman AE OE 2018',
-  'Harman without Bass Shelf',
-];
+  const { globalError, setGlobalError, performHealthCheck } = useAquaContext();
+  const [devices, setDevices] = useState<string[]>([]);
+  const [responses, setResponses] = useState<string[]>([]);
+  const [currentDevice, setCurrentDevice] = useState<string>('');
+  const [currentResponse, setCurrentResponse] = useState<string>('');
 
-const AutoEQ = () => {
-  const { globalError } = useAquaContext();
-  const [device, setDevice] = useState<string>(AUDIO_DEVICES[0]);
-  const [response, setResponse] = useState<string>(
-    TARGET_FREQUENCY_RESPONSE[0]
-  );
+  // Fetch supported devices from storage
+  useEffect(() => {
+    const fetchDeviceNames = async () => {
+      try {
+        setDevices(await getAutoEqDeviceList());
+      } catch (e) {
+        setGlobalError(e as ErrorDescription);
+      }
+    };
+
+    fetchDeviceNames();
+  }, [setGlobalError]);
+
+  // When user changes the current selected device, fetch the supported responses
+  useEffect(() => {
+    const fetchResponses = async () => {
+      try {
+        setResponses(await getAutoEqResponseList(currentDevice));
+        // Reset currentReponse to blank whenever it changes.
+        setCurrentResponse('');
+      } catch (e) {
+        setGlobalError(e as ErrorDescription);
+      }
+    };
+    // Skip checking for responses when we don't have a selected current device
+    if (currentDevice !== '') {
+      fetchResponses();
+    }
+  }, [currentDevice, setGlobalError]);
 
-  const applyAutoEQ = () => {
-    console.log('clicked');
+  const applyAutoEQ = async () => {
+    try {
+      await loadAutoEqPreset(currentDevice, currentResponse);
+      performHealthCheck();
+    } catch (e) {
+      setGlobalError(e as ErrorDescription);
+    }
   };
 
   const deviceOptions: IOptionEntry[] = useMemo(
     () =>
-      AUDIO_DEVICES.map((s) => {
+      devices.map((s) => {
         return {
           value: s,
           label: s,
           display: <div>{s}</div>,
         };
       }),
-    []
+    [devices]
   );
+
   const responseOptions: IOptionEntry[] = useMemo(
     () =>
-      TARGET_FREQUENCY_RESPONSE.map((s) => {
+      responses.map((s) => {
         return {
           value: s,
           label: s,
           display: <div>{s}</div>,
         };
       }),
-    []
+    [responses]
   );
+
   return (
     <div className="auto-eq">
       Audio Device:
       <Dropdown
         name="Audio Device"
         options={deviceOptions}
-        value={device}
-        handleChange={(newValue) => setDevice(newValue)}
+        value={currentDevice}
+        handleChange={(newValue) => setCurrentDevice(newValue)}
```

Comment:
Instead of updating `currentDevice` here and then fetching the corresponding responses in a `useEffect`, it could be more direct to simply do the fetch when the value is changed.

```typescript
const handleDeviceChange= async (newDevice) => {
  try {
    setResponses(await getAutoEqResponseList(newDevice));
    setCurrentDevice(newDevice);
    setCurrentResponse(''); // Unselect the current response whenever the device changes
  } catch (e) {
    setGlobalError(e as ErrorDescription);
  }
};
```

 ```HTML
<Dropdown
    ...
    handleChange={handleDeviceChange}
    ...
/>
```